PWGJE/EMCALJetTasks/macros/JetQA/scalePtHardHistos.py

```diff
@@ -134,24 +170,40 @@ def scalePtHardHistos(referenceFile):
       # Compute: scale factor = xsec per event / trials per event
       nEventsTot = hNEventsTotal.GetBinContent(bin+1)
       nEventsAcc = hNEventsAccepted.GetBinContent(bin+1)
-      xsec = hXsecPtHard.GetBinContent(1) / hXsecPtHard.GetEntries()
+      xsec = hXsecPtHard.GetBinContent(1) / hXsecPtHard.GetEntries() #entries in this case are number of files that were merged together
+      #print("ooo Test entries: {0}".format(hXsecPtHard.GetEntries()))
       trials = 1.*hTrialsPtHard.GetBinContent(1) / nEventsTot
       scaleFactor = xsec/trials
-      eventScaleFactor = nEventsAccAvg/nEventsAcc # also scale to account that there are different number of events in each Pt-hard bin
-
-      print "nEventsAcc: {0}".format(nEventsAcc)
-      print "nEventsTot: {0}".format(nEventsTot)
-      print "scaleFactor: {0}".format(scaleFactor)
-      print "eventScaleFactor: {0}".format(eventScaleFactor)
+      eventScaleFactorAcc = nEventsAccAvg/nEventsAcc # also scale to account that there are different number of events in each Pt-hard bin
+      eventScaleFactorTot = nNEventsTotAvg/nEventsTot
       
+      print("ooo nEventsAcc: {0}".format(nEventsAcc))
+      print("ooo nEventsTot: {0}".format(nEventsTot))
+      print("ooo nTrials: {0}".format(trials))
+      print("ooo scaleFactor: {0}".format(scaleFactor))
+      print("ooo eventScaleFactor Acc: {0}".format(eventScaleFactorAcc))
+      print("ooo eventScaleFactor All: {0}".format(eventScaleFactorTot))
+      print("ooo combined ScaleFactor: {0}".format(eventScaleFactorAcc*scaleFactor))
+      print("| *{0}* | {1} - {2} | {3} | {4} | {5} |".format((bin+1), ptHardLo[bin], ptHardHi[bin], scaleFactor, eventScaleFactorAcc, eventScaleFactorTot ),file=twikifile)
+
       hXSecPerEvent.Fill(bin+0.5, xsec)
       hNTrialsPerEvent.Fill(bin+0.5, trials)
       hScaleFactor.Fill(bin+0.5, scaleFactor)
 
       # Now, scale all the histograms
-      print "Scaling list: " + qaList.GetName()
+      print("ooo Scaling list: " + qaList.GetName())
+      RadiusName=""
       for obj in qaList:
-        ScaleAllHistograms(obj, scaleFactor * eventScaleFactor, f, bRemoveOutliers)
+        name = obj.GetName()
+        if "AKTFullR020" in name:
+          RadiusName="AKTFullR020"
+        if "AKTFullR030" in name:
+          RadiusName="AKTFullR030"
+        if "AKTFullR040" in name:
+          RadiusName="AKTFullR040"
```

Comment:
Can we just pass in a substring of the name, instead of hard-coding all these checks? For example, an easy way could be to pass in name[:15] to get the first 15 characters of the list name (or however many you need).